### ISSUE TITLE
LazyGit Seemingly Not Reading in Config

### ISSUE THREAD

**Issue by andscoop** [jesseduffield/lazygit#2406]
**Describe the bug**
I am attempting to add an enterprise github service to my lazygit config. After adding `services` config and still being unable to open a PR against the enterprise instance, I started testing other configurations to debug. It looks like none of those are being read in either.

My config and proof of config location
```
❯ cat /Users/john.smith/.config/lazygit/config.yml
gui:
  authorColors:
    'John Smith': 'green'
    '*': '#0000ff'
os:
  editCommand: 'vim'
services:
  'foo-github.com': 'github:foo-github.com'

❯ lazygit -cd
/Users/john.smith/.config/lazygit
```

**To Reproduce**

1. Check config directory with `lazygit -cd`
2. Create a new config in `$CONFIG_DIR/lazygit/config.yml`
3. Execut lazygit
4. Create PR, and check URL in PR is pointing to your enterprise github instance

**Expected behavior**
I'd expect that opening a PR uses the `services` config or errors if improperly configured.

**Version info:**
```
❯ lazygit --version
commit=, build date=, build source=homebrew, version=0.36.0, os=darwin, arch=amd64, git version=2.38.1
❯ git --version
git version 2.38.1
```

**Additional context**

As a side note-  I attempted to debug but the log file is not being created after running with `--debug`.

**Note:** please try updating to the latest version or [manually building](https://github.com/jesseduffield/lazygit/#manual) the latest `master` to see if the issue still occurs.

**Comment by andscoop** [comment]
I think I'm wrong about the config not working. I actually am seeing author initials reflected in the same colors I have set in this config.

I am still not able to open a PR against our GHE instance though with this config.

**Comment by A-Lamia** [comment]
I'm having this issue currently on windows just tried settings and nothing happens just opens normal VIM no hotkey changes are working as well.
```yml
os:
	editCommand: 'nvim'
	
```

**Comment by mark2185** [comment]
@A-Lamia try using two spaces instead of tabs.

**Comment by A-Lamia** [comment]
@mark2185 yeah still nothing sadly

**Comment by mark2185** [comment]
Could you post the path to your config, the entire config, and the output of running `lazygit -cd`?

**Comment by A-Lamia** [comment]
path: C:\Users\Lamia\AppData\Roaming\lazygit
lazygit -cd: C:\Users\Lamia\AppData\Roaming\lazygit

config:
```yaml
os:
  editCommand: 'nvim'
keybinding:
  appendNewLine: '<a-o>'

```

**Comment by mark2185** [comment]
Is `nvim` in the path? Try

`keybinding.appendNewLine` is sadly not used at all, try setting something that's visible right from the get go, e.g. the theme
```
gui:
  showIcons: true
  theme:
    activeBorderColor:
      - '#1eaeff'
      - bold
```

**Comment by A-Lamia** [comment]
that is interesting the color changed, wait how is lazygit running the commands ?

**Comment by mark2185** [comment]
It runs `cmd /c nvim` in the background.

**Comment by A-Lamia** [comment]
yeah im not sure i tried neovide, nvim, vscode all of the can be launched from both cmd and powershell but vim is what it keeps openin.

**Comment by mark2185** [comment]
That's odd, it first checks for [os.editCommand](https://github.com/jesseduffield/lazygit/blob/82fc6fb111ba94aaabf78a00396e69689b860655/pkg/commands/git_commands/file.go#L31), and only if that's empty does it check git's editor and whatnot.

What does the log (bottom right panel) say after opening a file? 
`cmd /c vim` regardless of what you put in there?

**Comment by A-Lamia** [comment]
open works fine, edit works fine but commit in editor is in VIM is this normal ?

**Comment by mark2185** [comment]
Yup, according to the source:
```
// runs git commit without the -m argument meaning it will invoke the user's editor
func (self *CommitCommands) CommitEditorCmdObj() oscommands.ICmdObj {
	return self.cmd.New(fmt.Sprintf("git commit%s%s", self.signoffFlag(), self.verboseFlag()))
}
```

**Comment by A-Lamia** [comment]
yeah i just got everything working LOL sorry for bothering you.

**Comment by A-Lamia** [comment]
I think for people like me who decide to leave there brain in the fridge when i woke up you should add something here https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md#configuring-file-editing

about commit editor being ran with gits set editor. :)

**Comment by mark2185** [comment]
Well, it's not _file editing_ per se, it's _committing_.

We could add this, but given the confusion regarding `edit` and `commit`, I doubt anyone would benefit from it:
```
    commitChanges: 'c' # commit with git's set editor
```
